lib/setupmanagers/hckclient.rb

```diff
@@ -134,6 +149,8 @@ def initialize_client_wait
     end
 
     def return_when_client_up
+      move_machine_to_default_pool if machine_in_working_pool?
```

Comment:
Why do we need this? Let's just skip recognize_client_wait and initialize_client_wait if client initialized and not in Default Pool.